### PR TITLE
Update LLVM, WAVM, LibP2P

### PR DESCRIPTION
**Pull request by Harrm**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/create/cmake.html)
  step by step carefully. **Hai.**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

https://github.com/soramitsu/kagome/pull/922/checks?check_run_id=3675932193 - doesn't pass because the libp2p is too new, but the hunter part works fine, the problem is in Kagome and is already covered by https://github.com/soramitsu/kagome/pull/923

* I have submitted CI configs to https://github.com/cpp-pm/hunter-testing targeting `pkg.template` branch: **Iie**.

---

---

### CODE REVIEW

File: cmake/projects/binaryen/hunter.cmake [hunk 1]
```diff
@@ -9,22 +9,34 @@ include(hunter_download)
 include(hunter_pick_scheme)
 include(hunter_cmake_args)
 
+
+hunter_add_version(
+    PACKAGE_NAME
+    binaryen
+    VERSION
+    1.38.28-p0
```
Comment (igor-egorov) [inline cmake/projects/binaryen/hunter.cmake:17]: Why do you put it here when it was not used and we already have p1?

File: cmake/projects/wavm/hunter.cmake [hunk 2]
```diff
@@ -15,9 +15,9 @@ hunter_add_version(
     VERSION
     1.0.0
     URL
-    "https://github.com/soramitsu/WAVM/archive/9b914111910e9ece1a9243923059d16c153f4c9f.zip"
+    "https://github.com/soramitsu/WAVM/archive/c7dbf7d7af445491bbe7e0b26ea8e4c7b300896a.tar.gz"
```
Comment (igor-egorov) [inline cmake/projects/wavm/hunter.cmake:18]: Please change the version number too when you are changing sources commit.

**Review by sanblch — APPROVED** [review]

File: cmake/projects/Clang/hunter.cmake [hunk 3]
```diff
@@ -7,6 +7,17 @@ include(hunter_add_version)
...
 
+hunter_add_version(
+    PACKAGE_NAME
+    Clang
+    VERSION
+    "12.0.1-p0"
+    URL
+    "https://github.com/soramitsu/kagome-clang/archive/v12.0.1-p0.tar.gz"
+    SHA1
+    b9507482e56b614ad3c92134eb2fc3749a38aaaa
```
Comment (igor-egorov) [inline cmake/projects/Clang/hunter.cmake:18]: f290cb45343bc9032823c82b1d021cc02c762acd

File: cmake/projects/LLVM/hunter.cmake [hunk 4]
```diff
@@ -11,6 +11,17 @@ include(hunter_download)
...
 
+hunter_add_version(
+    PACKAGE_NAME
+    LLVM
+    VERSION
+    "12.0.1-p1"
+    URL
+    "https://github.com/soramitsu/kagome-llvm/archive/v12.0.1-p1.tar.gz"
+    SHA1
+    33af04396eb2a184aa1dafe2fcb4ab3c7526795b
```
Comment (igor-egorov) [inline cmake/projects/LLVM/hunter.cmake:22]: 5c90f60281f73c9a4f8d8927c541e39ee43a04fc

File: cmake/projects/binaryen/hunter.cmake [hunk 5]
```diff
@@ -9,22 +9,34 @@ include(hunter_download)
...
+    URL
+    "https://github.com/soramitsu/binaryen/archive/1.38.28-p0.tar.gz"
```
Comment (igor-egorov) [inline cmake/projects/binaryen/hunter.cmake:19]: moreover - HTTP 404 !

File: cmake/projects/binaryen/hunter.cmake [hunk 6]
```diff
@@ -9,22 +9,34 @@ include(hunter_download)
...
+
 hunter_add_version(
     PACKAGE_NAME
     binaryen
     VERSION
-    version_101
+    1.38.28-p1
     URL
-    "https://github.com/soramitsu/binaryen/archive/refs/tags/version_101.tar.gz"
+    "https://github.com/soramitsu/binaryen/archive/1.38.28-p1.tar.gz"
```
Comment (igor-egorov) [inline cmake/projects/binaryen/hunter.cmake:30]: HTTP 404 too 🤔

**Review by igor-egorov — CHANGES_REQUESTED** [review]

**Comment by xDimon** [comment]
Please, update libp2p too.

File: cmake/projects/binaryen/hunter.cmake (hunk 1 above)
Comment (Harrm) [reply cmake/projects/binaryen/hunter.cmake:17]: Were trying to revert binary, as we found out that new versions have completely unfeasible performance, but it turned out there are weird discrepancies between soramitsu/binaryen and Warchant/binaryen repos. Better deal with it in a separate PR.

**Review by igor-egorov — APPROVED** [review]

**Pull request merged**